indra/sources/signor.py

```diff
@@ -0,0 +1,389 @@
+    'glycosylation': Glycosylation,
+    'hydroxylation': Hydroxylation,
+    'neddylation': None,
+    'chemical activation': Activation,
+    'chemical inhibition': Inhibition,
+    'trimethylation': Methylation,
+    'ubiquitination': Ubiquitination,
+    'post transcriptional regulation': None,
+    'relocalization': None, # TODO: Translocation,
+    'small molecule catalysis': None,
+    's-nitrosylation': None,
+    'transcriptional regulation': None,
+    'translation regulation': None,
+    'tyrosination': None,
+    'lipidation': None,
+    'oxidation': None,
+    'methylation': Methylation,
+    'palmitoylation': Palmitoylation,
+    'phosphorylation': Phosphorylation,
+    'stabilization': IncreaseAmount,
+    'sumoylation': Sumoylation
+}
+
+
+_effect_map = {
+    'down-regulates': Inhibition, # TODO: Need generic downregulation
+    'down-regulates activity': Inhibition,
+    'down-regulates quantity': DecreaseAmount,
+    'down-regulates quantity by destabilization': DecreaseAmount,
+    'down-regulates quantity by repression': DecreaseAmount,
+    'form complex': Complex,
+    'unknown': None,
+    'up-regulates': Activation, # TODO: Need generic upregulation
+    'up-regulates activity': Activation,
+    'up-regulates quantity': IncreaseAmount,
+    'up-regulates quantity by expression': IncreaseAmount,
+    'up-regulates quantity by stabilization': IncreaseAmount
+}
+
+
+class SignorProcessor(object):
+    """Processor for Signor dataset, available at http://signor.uniroma2.it.
+
+    Parameters
+    ----------
+    signor_csv : str
+        Path to SIGNOR CSV file. If None is given (default), the
+        SignorProcessor will attempt to download the full data file
+        from http://signor.uniroma2.it/downloads.php.
+    delimiter : str
+        Field delimiter for CSV file. Defaults to semicolon ';'.
+
+    Attributes
+    ----------
+    no_mech_rows: list of SignorRow namedtuples
+        List of rows where no mechanism statements were generated.
+    no_mech_ctr : collections.Counter
+        Counter listing the frequency of different MECHANISM types in the
+        list of no-mechanism rows.
+    """
+    def __init__(self, signor_csv=None, delimiter=';'):
+        # Get generator over the CSV file
+        if signor_csv:
+            data_iter = read_unicode_csv(signor_csv, delimiter=';', skiprows=1)
+            # Process into a list of SignorRow namedtuples
+            # Strip off any funky \xa0 whitespace characters
+            self._data = [SignorRow(*[f.strip() for f in r]) for r in data_iter]
+        # If no CSV given, download directly from web
+        else:
+            res = requests.post('http://signor.uniroma2.it/download_entity.php',
+                                data={'organism':'human', 'format':'csv',
+                                      'submit':'Download'})
+            if res.status_code == 200:
+                csv_reader = csv.reader(StringIO(res.text), delimiter=delimiter,
```

Comment:
The delimiter here needs to be bytes in Python 2. After trying to change that, line 168 gives a Unicode error.

Reply:
Fixed in d9d3ab32d.